packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts

```diff
@@ -376,6 +376,16 @@ describeNoCompat("stashed ops", (getTestObjectProvider) => {
 		assert.strictEqual(directory2.get(testKey), testValue);
 	});
 
+	it("connects when loaded with empty stash blob and no delta connection", async function () {
+		const pendingOps = await getPendingOps(provider, false);
+
+		const headers: IRequestHeader = { [LoaderHeader.loadMode]: { deltaConnection: "none" } };
+		const container2 = await loader.resolve({ url, headers }, pendingOps);
+		container2.connect();
+		await waitForContainerConnection(container2);
+		await provider.ensureSynchronized();
```

Comment:
Is it possible (in addition to what you have here) to create a UT for cases you are running into? ConnectionStateHandler has pretty elaborate set of UTs. Adding a case that demonstrates the problem might be helpful, as it's easier to see what is the actual sequence that is problematic, and if it's the sequence itself problematic, or the implementation of ConnectionStateHandler  (i.e. where the bugs are).

Reply:
I added a UT. This is the sequence of events that I see from debugging.

Before I made the change to force write mode when dirty, the same sequence would happen for non-empty stash blobs. In that case it's important that we actually do wait for the previous client to leave, for the same reasons as normal reconnect flow. If the stash blob is empty, waiting is not necessary, but ConnectionStateHandler has limited knowledge about this.